V1.4-Update use case in DG

ZhiWei1010 opened the pull request:
This commit updates the use cases in the DG
* Edit all use case to end at extensions instead of resume
* Add new use case for edit event, edit patient, find patient by keyword, sort, help, clear, exit.
* Standardise all use cases.

Fixes #249

ZhiWei1010 committed:
Docs: Add new use case for sort command in DG

ZhiWei1010 committed:
Docs: Add unique identification for all use cases in DG

ZhiWei1010 committed:
Docs: Update Use case 05 to be more specific

ZhiWei1010 committed:
Merge branch 'master' into v1.4-DG-update-use-case

ZhiWei1010 committed:
Docs: Update use case 01 to be more specific

ZhiWei1010 committed:
Docs: Update use case 01 to be more specific

Update the steps information to be more specific

ZhiWei1010 committed:
Docs: Update Use Case 01 - Add a patient

Added an intermediate step where patient sync will process nurse input.

ZhiWei1010 committed:
Docs: Update use case 01

Remove intermediate step of PatientSync process nurse input to be
consistent with other use cases.

ZhiWei1010 committed:
Docs: Update use cases in DG to standardise

Add new use cases for help, find patient by name, clear, exit,
edit event.

ZhiWei1010 committed:
Docs: Add new edit patient use case

Renumber all use cases.

Add new line for all preconditions.

choowengyan commented on docs/DeveloperGuide.md:
```diff
@@ -1066,54 +1066,116 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+  
+* 1b. Nurse's input contains invalid parameters value. 
 
+    * 1b1. PatientSync shows an error message. 
```
Perhaps can add one more use case where Nurse added patient with duplicated patient hospital ID?

ZhiWei1010 replied:
do you think it can be counted as invalid parameters value?

choowengyan replied:
ya I think should be can!

choowengyan commented on docs/DeveloperGuide.md:
```diff
@@ -1066,54 +1066,116 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+1. Nurse requests to list patients.
+1. PatientSync shows a list of patients.
+1. Nurse requests to edit a specific patient in the list.
```
Would it be better to phrase it as `Nurse requests to edit a specific patient's information/ field in the list`?

ZhiWei1010 replied:
good suggestion! changed

choowengyan commented on docs/DeveloperGuide.md:
```diff
@@ -1066,54 +1066,116 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+1. Nurse requests to list patients.
+1. PatientSync shows a list of patients.
+1. Nurse requests to edit a specific patient in the list.
+1. PatientSync edit the patient and displays success message.
```
Would it be better to phrase it as` PatientSync edits the patient information and displays success message` ?

ZhiWei1010 replied:
good suggestion! changed

choowengyan commented on docs/DeveloperGuide.md:
```diff
@@ -1066,54 +1066,116 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
   Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. Nurse's input is invalid command format.
```
Would be better if it is phrased to `Nurse's input command format is invalid`?

choowengyan commented on docs/DeveloperGuide.md:
```diff
@@ -1066,54 +1066,116 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+    Use case ends.
+
+* 3b. Nurse's input contains invalid patient id.
```
Would suggest to change -> Nurse's input contains invalid ~~patient id~~ `patientHospitalId` / `patient's hospital ID` (depending on the other use cases format)

ZhiWei1010 replied:
the patient ID here should be referring to the id in the patient list, thus not `patientHospitalId`

choowengyan replied:
Would it be better if we use index instead?

choowengyan commented on docs/DeveloperGuide.md:
```diff
@@ -1066,54 +1066,116 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+  * 3c1. PatientSync shows an error message.
+
+    Use case ends.
```
Would be great as well if we could add one more use case where the index to edit inputted by nurse is invalid? or do you think it is considered under the invalid command format use case?

ZhiWei1010 replied:
I think it can be considered under invalid parameters!

choowengyan commented on docs/DeveloperGuide.md:
```diff
@@ -1163,20 +1247,26 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+1. Nurse requests to list patients.
+1. PatientSync shows a list of patients.
+1. Nurse requests to add a tag to a specific patient in the list.
+1. PatientSync add a tag to the patient and displays success message.
```
Would be good if `add` change to `adds`

ZhiWei1010 replied:
good catch!

choowengyan reviewed (changes requested):
Overall, great use cases documentation! Just some suggestions on the use cases for add/ edit patient.

ZhiWei1010 committed:
Docs: Update use case after review from wengyan

guohuang88 commented on docs/DeveloperGuide.md:
```diff
@@ -1163,20 +1253,26 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+  * 3b1. PatientSync shows an error message.
+
+    Use case ends.
 
```
```suggestion
* 3c. Nurse's input contains invalid event name.
    * 3c1. PatientSync shows an error message.
    Use case resumes at step 2.
* 3d. Nurse's input contains invalid date or datetime format.
    * 3d1. PatientSync shows an error message.
    Use case resumes at step 2.
* 3e. Nurse's input contains duplicate event.
    * 3e1. PatientSync shows an error message.
    Use case resumes at step 2.
```
These are the other cases where there will be an error message.

ZhiWei1010 replied:
Updated!

ZhiWei1010 committed:
Docs: Update use case ends to use case resumes from step 2

Update use case to resume from step 2 instead of ends after offline
discussion for the use cases that lists the patients first.

choowengyan commented on docs/DeveloperGuide.md:
```diff
@@ -1066,54 +1066,122 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+    
+      Use case ends.
+
+* 3b. Nurse's input contains invalid patient id.
```
should it be `ID`/ `Id` instead if `id`?

ZhiWei1010 replied:
Good catch!

choowengyan commented on docs/DeveloperGuide.md:
```diff
@@ -1066,54 +1066,122 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+    Use case ends.
+
+* 3b. Nurse's input contains invalid patient id.
```
Similar to the comment above regarding `id`.

ZhiWei1010 replied:
Good catch!

choowengyan reviewed (commented):
Just some comments regarding id as they are different from the Id in line 1094. Perhaps we can consider standardising whether we want id or Id / ID

guohuang88 commented on docs/DeveloperGuide.md:
```diff
@@ -1163,20 +1224,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+1.  Nurse requests to list patients.
+1.  PatientSync shows a list of patients.
+1.  Nurse requests to edit an event for a specific patient in the list.
+1.  PatientSync edits an event the patient and displays success message.
```
```suggestion
1.  PatientSync edits an event for the patient and displays success message.
```

ZhiWei1010 replied:
Good catch! updated!

guohuang88 commented on docs/DeveloperGuide.md:
```diff
@@ -1232,12 +1315,123 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
       Use case resumes at step 2.
 
+**Use case: UC11 - Find patients by keyword**
```
```suggestion
**Use case: UC11 - Find patients by name**
```

ZhiWei1010 replied:
It is being consistent with the Section on locating patients by name 3.10. Maybe should standardise both?

ZhiWei1010 replied:
After offline discussion, it is ok to change to name!

guohuang88 commented on docs/DeveloperGuide.md:
```diff
@@ -1232,12 +1315,123 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+1. Nurse requests to list patients.
+1. PatientSync shows a list of patients.
+1. Nurse requests to find patients with specific keywords in the list.
+1. PatientSync finds patients with the specified keywords and displays success message.
```
```suggestion
1. Nurse requests to find patients with specific name(s) in the list.
1. PatientSync finds patients with the specified name(s) and displays success message.
```

ZhiWei1010 replied:
It is being consistent with the Section on locating patients by name 3.10. Maybe should standardise both?

ZhiWei1010 replied:
After offline discussion, it is ok to change to name!

guohuang88 commented on docs/DeveloperGuide.md:
```diff
@@ -1232,12 +1315,123 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+1. Nurse requests to list patients.
+1. PatientSync shows a list of patients.
+1. Nurse requests to find patients with a specific tag in the list.
+1. PatientSync finds patients with the specified tag and displays success message.
```
```suggestion
1. Nurse requests to find patients with specific tag(s) in the list.
1. PatientSync finds patients with the specified tag(s) and displays success message.
```

guohuang88 replied:
User can find using 1 or more tags

ZhiWei1010 replied:
Nice catch! updated!

guohuang88 commented on docs/DeveloperGuide.md:
```diff
@@ -1232,12 +1315,123 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+    Use case resumes at step 2.
+
+**Use case: UC10 - Find patients by a tag**
```
```suggestion
**Use case: UC10 - Find patients by tag**
```

ZhiWei1010 replied:
updated to tag(s)! to be consistent with below comments!

ZhiWei1010 committed:
Docs: Update some typo error in use cases in DG

guohuang88 reviewed (changes requested):
Good job on rewriting the use cases! I have added some suggestions!

NeoHW commented on docs/DeveloperGuide.md:
```diff
@@ -1186,20 +1255,26 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
-4.  PatientSync deletes a tag from the patient
+1. Nurse requests to list patients.
+1. PatientSync shows a list of patients.
+1. Nurse requests to add a tag to a specific patient in the list.
```
one or more tags

ZhiWei1010 replied:
Updated !

NeoHW commented on docs/DeveloperGuide.md:
```diff
@@ -1186,20 +1255,26 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+1. Nurse requests to list patients.
+1. PatientSync shows a list of patients.
+1. Nurse requests to add a tag to a specific patient in the list.
+1. PatientSync adds a tag to the patient and displays success message.
```
adds the specified tag(s) and display status message

ZhiWei1010 replied:
Updated!

NeoHW commented on docs/DeveloperGuide.md:
```diff
@@ -1209,20 +1284,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+**Use case: UC09 - Delete tag from a patient**
+
+**Preconditions: Tag exists for the target patient**
```
better not to use preconditions in use case and put it under extensions

ZhiWei1010 replied:
Good point! removed the precondition!

NeoHW commented on docs/DeveloperGuide.md:
```diff
@@ -1232,12 +1315,123 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+  * 3c1. PatientSync shows an error message.
+
+    Use case resumes at step 2.
+
```
* 3d. The patient selected does not contain specified tag
* 3d1. PatientSync shows an error message.
Use case resumes at step 2

ZhiWei1010 replied:
Updated!

NeoHW commented on docs/DeveloperGuide.md:
```diff
@@ -1209,20 +1284,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
-4.  PatientSync finds patients with the tag
+1. Nurse requests to list patients.
+1. PatientSync shows a list of patients.
+1. Nurse requests to delete a tag from a specific patient in the list.
```
one or more tags

ZhiWei1010 replied:
Updated!

NeoHW commented on docs/DeveloperGuide.md:
```diff
@@ -1209,20 +1284,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+1. Nurse requests to list patients.
+1. PatientSync shows a list of patients.
+1. Nurse requests to delete a tag from a specific patient in the list.
+1. PatientSync deletes a tag from the patient and displays success message.
```
delete the specified tag(s) in the patient's tag list

ZhiWei1010 replied:
Updated!

ZhiWei1010 committed:
Docs: Update Use Cases After Review

Add new use extension to relevant use cases when patient specified
does not exist in patient list.

Remove all preconditions and add them under extensions instead for
better clarity.

choowengyan reviewed (approved):
LGTM!

guohuang88 reviewed (commented):
LGTM

ZhiWei1010 committed:
Merge branch 'master' into v1.4-DG-update-use-case

NeoHW reviewed (approved):
lgtm

SherisseTJW reviewed (approved):
lgtm
